contracts/components/staking/SlashingController.sol

```diff
@@ -0,0 +1,415 @@
+        address proposer,
+        uint256 subjectId,
+        uint8 subjectType,
+        bytes32 penaltyId
+    );
+    event EvidenceSubmitted(uint256 proposalId, StateMachines.State stateId, string[] evidence);
+    event SlashingExecutorChanged(address indexed slashingExecutor);
+    event StakingParametersManagerChanged(address indexed stakingParametersManager);
+    event DepositAmountChanged(uint256 amount);
+    event SlashPercentToProposerChanged(uint256 amount);
+    event DepositSubmitted(uint256 indexed proposalId, address indexed proposer, uint256 amount);
+    event DepositReturned(uint256 indexed proposalId, address indexed proposer, uint256 amount);
+    event DepositSlashed(uint256 indexed proposalId, address indexed proposer, uint256 amount);
+    event SlashPenaltyAdded(bytes32 indexed penaltyId, uint256 percentSlashed, PenaltyMode mode);
+    event SlashPenaltyRemoved(bytes32 indexed penaltyId, uint256 percentSlashed, PenaltyMode mode);
+
+    error WrongSlashPenaltyId(bytes32 penaltyId);
+    error NonExistentProposal(uint256 proposalId);
+    error NonRegisteredSubject(uint8 subjectType, uint256 subjectId);
+    error WrongPercentValue(uint256 value);
+
+    modifier onlyValidSlashPenaltyId(bytes32 penaltyId) {
+        if (penalties[penaltyId].mode == PenaltyMode.UNDEFINED) revert WrongSlashPenaltyId(penaltyId);
+        _;
+    }
+
+    modifier onlyValidPercent(uint256 percent) {
+        if (percent > 100) revert WrongPercentValue(percent);
+        _;
+    }
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(address _forwarder, address _depositToken) initializer ForwardedContext(_forwarder) {
+        if (_depositToken == address(0)) revert ZeroAddress("_depositToken");
+        depositToken = IERC20(_depositToken);
+        _transitiontable = StateMachines
+            .EMPTY_MACHINE
+            .addEdgeTransition(UNDEFINED, CREATED)
+            .addEdgeTransition(CREATED, DISMISSED)
+            .addEdgeTransition(CREATED, REJECTED)
+            .addEdgeTransition(CREATED, IN_REVIEW)
+            .addEdgeTransition(IN_REVIEW, REVIEWED)
+            .addEdgeTransition(IN_REVIEW, REVERTED)
+            .addEdgeTransition(REVIEWED, EXECUTED)
+            .addEdgeTransition(REVIEWED, REVERTED);
+    }
+
+    /**
+     * @notice Initializer method, access point to initialize inheritance tree.
+     * @param __manager address of AccessManager.
+     * @param __router address of Router.
+     */
+    function initialize(
+        address __manager,
+        address __router,
+        ISlashingExecutor __executor,
+        FortaStakingParameters __stakingParameters,
+        uint256 __depositAmount,
+        uint256 __slashPercentToProposer,
+        bytes32[] calldata __slashPenaltyIds,
+        SlashPenalty[] calldata __slashPenalties
+    ) public initializer {
+        __AccessManaged_init(__manager);
+        __Routed_init(__router);
+        __UUPSUpgradeable_init();
```

Comment:
If I understand correctly these come from `BaseComponentUpgradeable`. I would recommend adding a function `__BaseComponentUpgradeable_init(__manager, __router)` and using that instead.

Reply:
Noted, I will add it post security review (since this should affect all contracts inheriting from `BaseComponentUpgradeable`, better include it in other PR before the main audit.)